### PR TITLE
ra-multiplex: 0.2.2 -> 0.2.5

### PR DESCRIPTION
**Pull request by Congee**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by NorfairKing — APPROVED** [review]
LGTM

**Comment by Congee** [comment]
Result of `nixpkgs-review pr 344462` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ra-multiplex</li>
  </ul>
</details>

**Comment by Congee** [comment]
A caveat is the user needs to do `pass_environment = ["PATH"]` if ra-multiplex is run by a service like systemd/launchd

File: pkgs/development/tools/rust/ra-multiplex/default.nix [hunk 1]
```diff
@@ -5,24 +5,24 @@
...
   pname = "ra-multiplex";
-  version = "0.2.2";
+  version = "0.2.5";
 
   src = fetchFromGitHub {
     owner = "pr2502";
-    repo = "ra-multiplex";
-    rev = "dcb5f83890cb91016b0a1590cc1b732606bb6ec1";
-    hash = "sha256-Hf4Gj9eXEP4gXiqNV4Jq0oiGLX3DtDF9At1feEZ+bUE=";
+    repo = pname;
```
Comment (pbsds) [inline pkgs/development/tools/rust/ra-multiplex/default.nix:14]:
```suggestion
    repo = "ra-multiplex";
```

Using pname is in general considered an anti-pattern, please undo

Comment (Congee) [reply pkgs/development/tools/rust/ra-multiplex/default.nix]: done. would you mind giving a pointer about why it is anti-pattern?

Comment (pbsds) [reply pkgs/development/tools/rust/ra-multiplex/default.nix]: It is tied to override interface. Changing the store path pname should not break the download location

**Review by pbsds — APPROVED** [review]

File: pkgs/development/tools/rust/ra-multiplex/default.nix (hunk 1 above)
Comment (pbsds) [reply pkgs/development/tools/rust/ra-multiplex/default.nix]: https://github.com/NixOS/nixpkgs/issues/277994

**Review by h7x4 — APPROVED** [review]

File: pkgs/development/tools/rust/ra-multiplex/default.nix (hunk 1 above)
Comment (Congee) [reply pkgs/development/tools/rust/ra-multiplex/default.nix]: thank you very much!

**Pull request merged**